test/internals/reply.test.js

```diff
@@ -293,6 +293,41 @@ test('buffer with content type should not send application/octet-stream', t => {
   })
 })
 
+test('stream using reply.res.writeHead should return customize headers', t => {
```

Comment:
The test confuses me a bit: no header are set, so there's no reason for call the warn.

The truth is that fastify adds an header internally here
https://github.com/fastify/fastify/blob/master/lib/reply.js#L57-L63

And that is why the warn is raised.

Maybe adding an explicit header using `res.header`  method, the test becomes more readable

Reply:
I have separate my test case about stream mode, I think it will be more clearly.   

`res.setHeader` won't dispatch the else branch, only for `res.writeHead`.  If `res.writeHead` has called before `res.setHeader`and don't have a judgment about `res.headersSent`, it will throw an error.  And that is the reason why I add this judgment, It can stop the error happening.